addon/components/base/bs-modal/dialog.js

```diff
@@ -204,18 +224,34 @@ export default class ModalDialog extends Component {
   }
 
   _click(e) {
+    if (this.ignoreBackdropClick) {
+      this.set('ignoreBackdropClick', false);
+      return;
+    }
     if (e.target !== this.element || !this.get('backdropClose')) {
       return;
     }
     this.get('onClose')();
   }
 
+  _mouseDown(e) {
+    this.set('mouseDownElement', e.target);
+  }
+
+  _mouseUp(e) {
+    if (this.mouseDownElement !== this.element && e.target === this.element) {
+      this.set('ignoreBackdropClick', true);
+    }
+  }
+
   didInsertElement() {
     super.didInsertElement(...arguments);
     // Ember events use event delegation, but we need to add an `onclick` handler directly on the modal element for
     // iOS to allow clicking the div. So a `click(){}` method here won't work, we need to attach an event listener
     // directly to the element
     this.element.onclick = bind(this, this._click);
+    this.element.onmousedown = bind(this, this._mouseDown);
+    this.element.onmouseup = bind(this, this._mouseUp);
```

Comment:
The explicit `onclick` handler was required for iOS issue described in the comment. But the other ones could use just the built-in event handler methods (https://guides.emberjs.com/v3.14.0/components/handling-events/), i.e. rename `_mouseDown` to `mouseDown` and not have this code in `didInsertElement`. Or do I miss something?

Reply:
Yeah, you are right about the event handlers, I updated pr with ember's default event handlers, thank you for warning me about this.